### PR TITLE
Replaces /client with /datum/client_base

### PR DESCRIPTION
**Pull request by Cyberboss**
There will soon come a day where there will be need to fake the presence of clients. Given that we can't instantiate /client directly, this is the next best thing.

Using byond fuckery, this folds everything about clients that we use into a type that can be instantiated. Added .tgclient var to mobs, which should be used instead of .client. It's set in Login() and Logout().

Main changes in modules/client/client_definitions.dm and modules/client/client_procs.dm

Implementation workflow:
```
/client -> /datum/client_base (auto)
client/ -> datum/client_base/ (find and replace)
```

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -109,138 +109,141 @@ GLOBAL_LIST_INIT(admin_verbs_server, world.AVerbsServer())
...
+	/datum/tgclient/proc/debug_huds,
+	/datum/tgclient/proc/customiseSNPC,
+	/datum/tgclient/proc/resetSNPC,
 	))
 
-/client/proc/add_admin_verbs()
+/client/add_admin_verbs()
 	if(holder)
 		control_freak = CONTROL_FREAK_SKIN | CONTROL_FREAK_MACROS
+	return ..()
```
Comment (Cyberboss) [inline code/modules/admin/admin_verbs.dm]: This one proc is an oddity because overriding vars which you set an compile time can be weird

**Comment by Fox-McCloud** [comment]
What could *possibly* go wrong?

**Comment by outbools** [comment]
Well you need a bit of preparation when something happens with ancient semi-maintained platform anyway

**Comment by Cyberboss** [comment]
Removed the changes to .client for now. Probably will do that in a separate PR with help from @RemieRichards

**Comment by RemieRichards** [comment]
🎩 
👀
 ☕️ 👌

**Comment by Cyberboss** [comment]
Found a better regex for client btw: `(^|&|\||=|\(|\[|\s|\.|\?|:)(client)(&|\||=|\(|\[|\s|\.|\?|:|$)`

**Comment by Tacolizard** [comment]
>There will soon come a day where there will be need to fake the presence of clients. 

🤔

**Comment by RemieRichards** [comment]
Y.A.L.P.E.R

**Comment by Jordie0608** [comment]
And maybe on that day we'll learn what it actually is.

**Comment by Cyberboss** [comment]
#40000

**Comment by RemieRichards** [comment]
the people who need to know already know.

**Comment by Jordie0608** [comment]
Elucidating.

**Comment by ike709** [comment]
Surprisingly, only one conflicting file.

**Comment by optimumtact** [comment]
Get this moving, I'm not a fan of your project already and I'm not willing to be patient

**Comment by Cyberboss** [comment]
I'll try this again when I'm less swamped

**Comment by ma44** [comment]
#40000

**Pull request closed without merging**